echopype/convert/ek60.py

```diff
@@ -304,131 +329,108 @@ def _set_nmea_dict():
+                               "gain_correction": "gain"}
+            param_str = {"gpt_software_version": "gpt_software_version",
+                         "channel_id": "channel_id",
+                         "beam_type": "beam_type"}
+
+            for encode_name, origin_name in param_numerical.items():
+                beam_dict[encode_name] = np.array(
+                    [val[origin_name] for key, val in self.config_datagram['transceivers'].items()]).astype('float32')
+            beam_dict['transducer_offset_z'] += [self.ping_data_dict[x]['transducer_depth'][0]
+                                                 for x in self.config_datagram['transceivers'].keys()]
+
+            for encode_name, origin_name in param_str.items():
+                beam_dict[encode_name] = [val[origin_name]
+                                          for key, val in self.config_datagram['transceivers'].items()]
+
+            beam_dict['transmit_signal'] = self.tx_sig[piece_seq]  # only this range_bin group
+
             # Build other parameters
             beam_dict['non_quantitative_processing'] = np.array([0, ] * freq.size, dtype='int32')
             # -- sample_time_offset is set to 2 for EK60 data, this value is NOT from sample_data['offset']
             beam_dict['sample_time_offset'] = np.array([2, ] * freq.size, dtype='int32')
 
-            idx = [np.argwhere(np.isclose(tx_sig['transmit_duration_nominal'][x - 1],
+            idx = [np.argwhere(np.isclose(self.tx_sig[piece_seq]['transmit_duration_nominal'][x - 1],
                                           self.config_datagram['transceivers'][x]['pulse_length_table'])).squeeze()
                    for x in self.config_datagram['transceivers'].keys()]
             beam_dict['sa_correction'] = \
                 np.array([x['sa_correction_table'][y]
                           for x, y in zip(self.config_datagram['transceivers'].values(), np.array(idx))])
 
-            return beam_dict
+            # New path created if the power data is broken up due to varying range bins
+            if len(self.range_lengths) > 1:
+                beam_dict['path'] = self.all_files[piece_seq]
+            else:
+                beam_dict['path'] = self.save_path
 
-        # Load data from RAW file
-        if not bool(self.power_dict):  # if haven't parsed .raw file
-            self.load_ek60_raw()
+            return beam_dict
 
-        # Get exported filename
-        filename = os.path.splitext(os.path.basename(self.filename))[0]
-        self.save_path = os.path.join(os.path.split(self.filename)[0], filename + file_format)
-        self.nc_path = os.path.join(os.path.split(self.filename)[0], filename + '.nc')
-        self.zarr_path = os.path.join(os.path.split(self.filename)[0], filename + '.zarr')
+        def copyfiles():
+            self.all_files = []
+            split = os.path.splitext(self.save_path)
+            for piece in range(len(self.range_lengths)):
```

Comment:
this `piece` shadows the same name from outer scope, better to change to something else to avoid confusion